src/toast/ops/hwpss_model.py

```diff
@@ -0,0 +1,936 @@
+                ch_start = 0
+                for ch in range(non_overlap):
+                    ch_mid = ch_start + half_chunk
+                    chunks.append(
+                        {
+                            "start": ch_start,
+                            "end": ch_start + chunk_samples,
+                            "time": reltime[ch_mid],
+                        }
+                    )
+                    if ch != non_overlap - 1:
+                        # Add the overlapping chunk
+                        chunks.append(
+                            {
+                                "start": ch_start + half_chunk,
+                                "end": ch_start + half_chunk + chunk_samples,
+                                "time": reltime[ch_start + chunk_samples],
+                            }
+                        )
+                    ch_start += chunk_samples
+        else:
+            # Use the specified interval list for the chunks.  Cut any
+            # chunks that are tiny.
+            for intr in obs.intervals[self.chunk_view]:
+                ch_size = intr.last - intr.first + 1
+                ch_mid = intr.first + ch_size // 2
+                if ch_size > 10 * self.harmonics:
+                    chunks.append(
+                        {
+                            "start": intr.first,
+                            "end": intr.last + 1,
+                            "time": reltime[ch_mid],
+                        }
+                    )
+        return chunks
+
+    def _compute_flags(self, obs, dets):
+        # The shared flags
+        if self.shared_flags is None:
+            shared_flags = np.zeros(obs.n_local_samples, dtype=np.uint8)
+        else:
+            shared_flags = np.array(obs.shared[self.shared_flags].data)
+            shared_flags &= self.shared_flag_mask
+
+        # Compute flags for samples where the hwp is stopped
+        stopped = self._stopped_flags(obs)
+        shared_flags |= stopped
+
+        # If we are chunking based on intervals, flag the regions between valid
+        # intervals.
+        if self.chunk_view is not None:
+            not_modelled = np.ones_like(shared_flags)
+            for intr in obs.intervals[self.chunk_view]:
+                not_modelled[intr.first : intr.last + 1] = 0
+            shared_flags |= not_modelled
+
+        # Per-detector flags.  We merge in the shared flags to these since the
+        # detector flags will be written out at the end if the model is subtracted.
+        det_flags = dict()
+        for idet, det in enumerate(dets):
+            if self.det_flags is None:
+                det_flags[det] = shared_flags
+            else:
+                det_flags[det] = np.copy(obs.detdata[self.det_flags][det])
+                det_flags[det] &= self.det_flag_mask
+                det_flags[det] |= shared_flags
+        return (shared_flags, det_flags)
+
+    def _stopped_flags(self, obs):
```

Comment:
If the observation contains a time when the HWP is stopped, we should also flag deceleration/acceleration.

Reply:
I added code to detect and flag acceleration regions of the hwp angle.  The "basic" unit test now checks that case.